src/layer/riscv/absval_riscv.cpp

```diff
@@ -0,0 +1,161 @@
+// Xavier Hsinyuan is pleased to support the open source community by making ncnn available.
+//
+// Copyright (C) 2021 Xavier Hsinyuan <thelastlinex@hotmail.com>. All rights reserved.
+//
+// Licensed under the BSD 3-Clause License (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// https://opensource.org/licenses/BSD-3-Clause
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+#include "absval_riscv.h"
+
+#if __riscv_vector
+#ifdef RVV_SPEC_0_7
+#include "riscv_v_071_fix.h"
+#else
+#include <riscv_vector.h>
+#endif
+#include "rvv_mathfun.h"
+#include "rvv_mathfun_fp16s.h"
+
+static inline vfloat32m8_t vfabs_v_f32m8_absval(vfloat32m8_t op1, size_t vl)
+{
+    return vfsgnjx_vv_f32m8(op1, op1, vl);
+}
+static inline vfloat16m8_t vfabs_v_f16m8_absval(vfloat16m8_t op1, size_t vl)
+{
+    return vfsgnjx_vv_f16m8(op1, op1, vl);
+}
+#endif // __riscv_vector
+
+namespace ncnn {
+
+AbsVal_riscv::AbsVal_riscv()
+{
+#if __riscv_vector
+    support_packing = true;
+#if __riscv_zfh
+    support_fp16_storage = true;
+#endif
+#endif // __riscv_vector
+}
+
+int AbsVal_riscv::forward_inplace(Mat& bottom_top_blob, const Option& opt) const
+{
+    int elembits = bottom_top_blob.elembits();
+
+#if __riscv_vector && __riscv_zfh
+    if (opt.use_fp16_storage && elembits == 16)
+    {
+        if (opt.use_fp16_arithmetic)
+            return forward_inplace_fp16sa(bottom_top_blob, opt);
+        else
+            return forward_inplace_fp16s(bottom_top_blob, opt);
+    }
```

Comment:
For operators that has effectively precision difference caused by the intermediate accumulator type, we implement the two variants, fp16 and fp16sa.
For abs and relu, it is fine to implement only one fp16s function that uses fp16 intermediate type.